lib/server/socket.ts

```diff
@@ -30,54 +47,104 @@ const socket = ({ server, loadedServerOpts }): SocketHandler => {
   };
 
   const websocket = new Primus(server, ioConfig);
-  // websocket.authorize(async (req, done) => {
-  //   const maskedToken = maskToken(
-  //     req.uri.pathname.replaceAll(/^\/primus\/([^/]+)\//g, '$1').toLowerCase(),
-  //   );
-  //   const authHeader = req.headers['authorization'];
+  if (loadedServerOpts.config.BROKER_SERVER_MANDATORY_AUTH_ENABLED) {
+    websocket.authorize(async (req, done) => {
+      const connectionIdentifier = req.uri.pathname
+        .replaceAll(/^\/primus\/([^/]+)\//g, '$1')
+        .toLowerCase();
+      const maskedToken = maskToken(connectionIdentifier);
+      const authHeader =
+        req.headers['Authorization'] ?? req.headers['authorization'];
+      const brokerClientId = req.headers['x-snyk-broker-client-id'] ?? null;
+      const role = req.headers['x-snyk-broker-client-role'] ?? null;
+      if (
+        (!authHeader ||
+          !authHeader.toLowerCase().startsWith('bearer') ||
+          !brokerClientId) &&
+        loadedServerOpts.config.BROKER_SERVER_MANDATORY_AUTH_ENABLED
+      ) {
+        logger.debug({ maskedToken }, 'request missing Authorization header');
+        done({
+          statusCode: 401,
+          authenticate: 'Bearer',
+          message: 'missing required authorization header',
+        });
+        return;
+      }
 
-  //   if (
-  //     (!authHeader || !authHeader.startsWith('Bearer')) &&
-  //     loadedServerOpts.config.BROKER_SERVER_MANDATORY_AUTH_ENABLED
-  //   ) {
-  //     logger.error({ maskedToken }, 'request missing Authorization header');
-  //     done({
-  //       statusCode: 401,
-  //       authenticate: 'Bearer',
-  //       message: 'missing required authorization header',
-  //     });
-  //     return;
-  //   }
+      const jwt = authHeader
+        ? authHeader.substring(authHeader.indexOf(' ') + 1)
+        : '';
+      if (
+        !jwt &&
+        loadedServerOpts.config.BROKER_SERVER_MANDATORY_AUTH_ENABLED
+      ) {
+        done({
+          statusCode: 401,
+          authenticate: 'Bearer',
+          message: 'Invalid JWT',
+        });
+        return;
+      } else {
+        logger.debug(
+          { maskedToken: maskToken(connectionIdentifier), brokerClientId },
+          `Validating auth for connection ${connectionIdentifier} client Id ${brokerClientId}, role ${role}`,
+        );
+        const credsCheckResponse = await validateBrokerClientCredentials(
+          authHeader,
+          brokerClientId,
+          connectionIdentifier,
+        );
+        if (!credsCheckResponse) {
+          logger.debug(
+            { maskedToken: maskToken(connectionIdentifier), brokerClientId },
+            `Denied auth for Connection ${connectionIdentifier} client Id ${brokerClientId}, role ${role}`,
+          );
+          done({
+            statusCode: 401,
+            authenticate: 'Bearer',
+            message: 'Invalid credentials.',
```

Comment:
nit: can we standardise on case and use of full stops please? In this file we have:
- `'missing required authorization header'`, 
- `'Invalid JWT'` and 
- `'Invalid credentials.'`

Reply:
Will address this in separate PR #922